packages/insured-bridge-relayer/src/canonical-bridge-adapters/OptimismBridgeAdapter.ts

```diff
@@ -0,0 +1,90 @@
+import winston from "winston";
+import { ZERO_ADDRESS } from "@uma/common";
+
+import { providers } from "ethers";
+
+import { getMessagesAndProofsForL2Transaction } from "@eth-optimism/message-relayer";
+import { predeploys, getContractDefinition } from "@eth-optimism/contracts";
+import { getDeployedContractArtifact } from "@eth-optimism/contracts/dist/contract-deployed-artifacts";
+
+import BridgeAdapterInterface from "./BridgeAdapterInterface";
+
+import Web3 from "web3";
+import type { TransactionType } from "@uma/common";
+
+export class OptimismBridgeAdapter implements BridgeAdapterInterface {
```

Comment:
Just OOC, do we typically test these chain-specific classes? I figure we don't just because we're directly using their library, but wanted to ask just in case.

Reply:
ye, not tested :) worst that can happen though is it errors or the tx reverts so low risk with that. testing is not worth the pain, given the setup complexities for each chain.